File: content/en/api/dashboards/code_snippets/api-dashboard-get_all.rb
require 'rubygems'
require 'dogapi'

api_key = '<YOUR_API_KEY>'
app_key = '<YOUR_APP_KEY>'

dog = Dogapi::Client.new(api_key, app_key)

# retrieve all screenboards (dashboard with `free` layout)
dog.get_all_boards('free')

# retrieve all timeboards (dashboard with `ordered` layout)
dog.get_all_boards('ordered')

# retrieve all custom dashboards (dashboard with `free` or `ordered` layout)
dog.get_all_boards()

File: content/en/api/dashboards/code_snippets/api-dashboard-get_all.py
from datadog import initialize, api

options = {
    'api_key': '<YOUR_API_KEY>',
    'app_key': '<YOUR_APP_KEY>'
}

initialize(**options)

# retrieve all screenboards (dashboard with `free` layout)
api.Dashboard.get_all(layout_type='free')

# retrieve all timeboards (dashboard with `ordered` layout)
api.Dashboard.get_all(layout_type='ordered')

# retrieve all custom dashboards (dashboard with `free` or `ordered` layout)
api.Dashboard.get_all()
